### ISSUE TITLE
机器学习理论梳理

### ISSUE THREAD

**Issue by smart022** [smart022/articles#22]
# 逻辑回归 softmax 最大熵 广义线性模型
[自然语言处理基础-最大熵模型](https://ilewseu.github.io/2018/05/13/%E6%9C%80%E5%A4%A7%E7%86%B5%E6%A8%A1%E5%9E%8B/)
[第三部分 广义线性模型 (Generalized Linear Models)](https://kivy-cn.github.io/Stanford-CS-229-CN/#/Markdown/cs229-notes1)
[一文搞懂交叉熵在机器学习中的使用，透彻理解交叉熵背后的直觉](https://blog.csdn.net/tsyccnh/article/details/79163834)
[通过逻辑回归的sigmoid函数把线性回归转化到[0,1]之间，这个值为什么可以代表概率？](https://www.zhihu.com/question/41647192/answer/216631564)

[简单易懂的softmax交叉熵损失函数求导](https://www.jianshu.com/p/c02a1fbffad6)
建议使用的符号，见netual network and deep learning
a= F ( w*a + b )
z = w*a + b

book: 神经网络与深度学习
# 实现
[logistic regression](https://blog.goodaudience.com/logistic-regression-from-scratch-in-numpy-5841c09e425f)

**Comment by smart022** [comment]
# GBDT
[梯度提升入门介绍(GBM)](https://vimsky.com/article/3877.html) [英文版](https://medium.com/mlreview/gradient-boosting-from-scratch-1e317ae4587d)
[机器学习算法中 GBDT 和 XGBOOST 的区别有哪些？](https://www.zhihu.com/question/41354392)
[gbdt的残差为什么用负梯度代替？](https://www.zhihu.com/question/63560633)
[梯度提升树中为什么说目标函数关于当前模型的负梯度是残差的近似值？](https://www.zhihu.com/question/60625492)
[GBM Paper Reading](http://xtf615.com/paper/GBM.html)
wepon大佬有专述

# xgb
[什么XGBoost效果更好，速度更快](https://blog.csdn.net/ruggier/article/details/83381679)
[XGBoost笔记 点出近似分裂算法](https://tianle.me/2018/01/09/xgboost-note/)
[XGBoost特点、调参、讨论](https://blog.csdn.net/niaolianjiulin/article/details/76574216)
[xgboost是用二阶泰勒展开的优势在哪？](https://www.zhihu.com/question/61374305)
[XGBoost Loss function Approximation With Taylor Expansion](https://stats.stackexchange.com/questions/202858/xgboost-loss-function-approximation-with-taylor-expansion)
# Lgb
[微软官方介绍](https://www.msra.cn/zh-cn/news/features/lightgbm-20170105)
[『 论文阅读』LightGBM原理-LightGBM: A Highly Efficient Gradient Boosting Decision Tree](https://blog.csdn.net/shine19930820/article/details/79123216)
[lightGBM原理、改进简述](https://blog.csdn.net/niaolianjiulin/article/details/76584785)
[Lightgbm如何处理类别特征？](https://blog.csdn.net/anshuai_aw1/article/details/83275299)

**Comment by smart022** [comment]
# 正则
[机器学习中的范数规则化之（一）L0、L1与L2范数](https://blog.csdn.net/zouxy09/article/details/24971995)
[深入理解L1、L2正则化](https://zhuanlan.zhihu.com/p/29360425)
